File: Music.Short.Android/app/src/main/java/com/zitherharpmusic/zhmshort/ui/home/HomeFragment.java
package com.zitherharpmusic.zhmshort.ui.home;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.viewpager2.widget.ViewPager2;

import com.google.android.material.tabs.TabLayout;
import com.zitherharpmusic.zhmshort.MainActivity;
import com.zitherharpmusic.zhmshort.R;

public class HomeFragment extends Fragment {

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_home, container, false);
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        TabLayout tabLayout = view.findViewById(R.id.tab_layout);
        ViewPager2 viewPager = view.findViewById(R.id.view_pager);
        HomeAdapter homeAdapter = new HomeAdapter(this, ((MainActivity) requireActivity()).getUser());
        viewPager.setAdapter(homeAdapter);
        homeAdapter.attach(tabLayout, viewPager);
    }
}

File: Music.Client.Android/Kotlin/Music/short/src/main/java/com/zitherharp/music/shorts/ui/home/HomeFragment.kt
package com.zitherharp.music.shorts.ui.home

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import com.zitherharp.music.model.Short
import com.zitherharp.music.shorts.databinding.FragmentHomeBinding
import com.zitherharp.music.shorts.ui.shorts.ShortFullscreenFragment
import com.zitherharp.music.ui.adapter.FragmentStateAdapter

class HomeFragment : Fragment() {
    private var _binding: FragmentHomeBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        _binding = FragmentHomeBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        with (binding) {
            HomeAdapter(this@HomeFragment,
                arrayOf("Theo dõi", "Đề xuất")).attach(tabLayout, viewPager, 1)
        }
    }

    inner class HomeAdapter(fragment: Fragment,
                            tabNames: Array<String>): FragmentStateAdapter(fragment, tabNames) {
        override fun createFragment(position: Int): Fragment {
            when (position) {
                0 -> return ShortFullscreenFragment(Short.repository.values.drop(120))
                1 -> return ShortFullscreenFragment(Short.repository.values.shuffled().subList(0, 50))
            }
            TODO("Index $position out of bounds for ${tabNames.size}")
        }
    }
}